File: src/models/show.ts
export default class Show {
  id: string;
  title: string;
  date: Date;
  creator: string;

  constructor(id: string, title: string, date: string, creator: string) {
    this.id = id;
    this.title = title;
    this.date = new Date(date);
    this.creator = creator;
  }

  static fromJson(r: any): Show {
    return new Show(
      r.id,
      r.title,
      r.date,
      r.creator);
  }
};

File: mobile/lib/src/show_feature/show.dart
/// A placeholder class that represents an entity or model.
class Show {
  final String id;
  final String title;
  final DateTime date;

  const Show({required this.id, required this.title, required this.date});

  factory Show.fromJson(Map<String, dynamic> json) {
    return Show(
      id: json['id'],
      title: json['title'],
      date: DateTime.parse(json['date'].toString()),
    );
  }
}
